### PR TITLE
Filter pharmacy option if already in the list

### PR DESCRIPTION
**Pull request by jknapp25**
For top ranked pharmacies (Costco, Walgreens) we were potentially showing them twice in the list because they are two separate query results merged together. This dedupes.

### CODE REVIEW

File: apps/patient/src/utils/general.ts [hunk 1]
```diff
@@ -106,6 +106,23 @@ function isCloseEvent(event: types.PharmacyEvent): event is types.PharmacyCloseE
   return event.type === 'close';
 }
 
+/**
+ * Dedupes pharmacy options when merging new and existing pharmacies
+ * @param newPharmacies Pharmacies query result
+ * @param existingPharmacies Pre-loaded pharmacies
+ * @returns Deduped list of enriched pharmacy options
+ */
+export const preparePharmacyOptions = (
```
Comment (samkotlove) [inline apps/patient/src/utils/general.ts:115]: feel like `dedupePharmacyOptions` is a better name?

File: apps/patient/src/views/Pharmacy.tsx [hunk 2]
```diff
@@ -267,7 +265,10 @@ export const Pharmacy = () => {
       return;
     }
 
-    const preparedPharmacies: PharmacyWithHours[] = pharmaciesResult.map(preparePharmacy);
+    const preparedPharmacies: PharmacyWithHours[] = preparePharmacyOptions(
+      pharmaciesResult,
+      topRankedPharmacies
+    );
```
Comment (samkotlove) [inline apps/patient/src/views/Pharmacy.tsx:271]: is it the top ranked pharmacies would be duplicated?

File: apps/patient/src/utils/general.ts [hunk 3]
```diff
@@ -106,6 +106,23 @@ function isCloseEvent(event: types.PharmacyEvent): event is types.PharmacyCloseE
...
+  newPharmacies: types.Pharmacy[],
+  existingPharmacies: EnrichedPharmacy[] = []
+) => {
+  const existingPharmacyIds = new Set(existingPharmacies.map((p) => p.id));
+  const enrichedNewPharmacies = newPharmacies
+    .filter((p) => !existingPharmacyIds.has(p.id))
+    .map(preparePharmacy);
+  return [...existingPharmacies, ...enrichedNewPharmacies];
+};
```
Comment (samkotlove) [inline apps/patient/src/utils/general.ts:124]: could we get a unit test on this func?

File: apps/patient/src/utils/general.ts (hunk 1 above)
Comment (jknapp25) [reply apps/patient/src/utils/general.ts:115]: It also processes hours info

File: apps/patient/src/views/Pharmacy.tsx (hunk 2 above)
Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx:271]: yes

**Comment by marisbest2** [comment]
Though we should copy over the tests

**Pull request closed without merging**